src/style-manager.js

```diff
@@ -250,59 +250,70 @@ module.exports = class StyleManager {
 
 function transformDeprecatedShadowDOMSelectors (css, context) {
   const transformedSelectors = []
-  const transformedSource = postcss.parse(css)
-  transformedSource.walkRules((rule) => {
-    const transformedSelector = selectorParser((selectors) => {
-      selectors.each((selector) => {
-        const firstNode = selector.nodes[0]
-        if (context === 'atom-text-editor' && firstNode.type === 'pseudo' && firstNode.value === ':host') {
-          const atomTextEditorElementNode = selectorParser.tag({value: 'atom-text-editor'})
-          firstNode.replaceWith(atomTextEditorElementNode)
-        }
+  let transformedSource
+  try {
```

Comment:
We could wrap the whole function inside a try/catch but that would prevent v8 from optimizing it out. Given that we run this code during startup, I think that narrowing the scope of the unoptimized code that gets executed is a good idea.